### PR TITLE
Add Build Pipeline and Infrastructure

### PR DESCRIPTION
**Pull request by smneal**
- Add Azure Pipelines YAML
- Add base inf template

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    value: "SkillsFundingAgency_das-roatp-assessor"
+  - group: 'Prod Management Resources'    
+  - name: buildConfiguration
+    value: 'release'
+  - name: buildPlatform
+    value: 'anycpu'
+
+pool:
+  name: 'DAS - Continuous Integration'
+  demands:
```
Comment (jack-education) [inline azure-pipelines.yml:22]: Are these demands needed?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+resources:
+  repositories:
+  - repository: das-platform-building-blocks
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/heads/master
+    endpoint: "GitHub (SFA)"  
+
+steps:
+- task: gittools.gitversion.gitversion-task.GitVersion@5
```
Comment (jack-education) [inline azure-pipelines.yml:32]: That task has recently been deprecated and we now use UseGitVersion https://github.com/SkillsFundingAgency/das-content-api/blob/master/azure-pipelines.yml#L31

File: azure-pipelines.yml (hunk 1 above)
Comment (smneal) [reply azure-pipelines.yml]: Removed

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                    "keyVaultCertificateName": {
+                        "value": "[parameters('certificateName')]"
+                    },
+                    "keyVaultName": {
+                        "value": "[parameters('sharedKeyVaultName')]"
+                    },
+                    "keyVaultResourceGroup": {
+                        "value": "[parameters('sharedKeyVaultResourceGroupName')]"
+                    },
+                    "serverFarmId": {
```
Comment (jack-education) [inline azure/template.json:92]: This property isn't needed

File: azure-pipelines.yml (hunk 2 above)
Comment (smneal) [reply azure-pipelines.yml]: Updated

File: azure/template.json [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    "appServiceName": {
+                        "value": "[variables('appServiceName')]"
+                    },
+                    "appServicePlanName": {
+                        "value": "[parameters('sharedFrontendAppServicePlanName')]"
+                    },
+                    "appServicePlanResourceGroup": {
+                        "value": "[parameters('sharedAppServicePlanResourceGroup')]"
+                    },
+                    "appServiceAppSettings": {
```
Comment (jack-education) [inline azure/template.json:115]: Could the extra application settings used in https://github.com/SkillsFundingAgency/das-reservations-api/blob/master/azure/template.json#L166-L197 be used, worth discussion whether the Redis connection string should also be within the application settings

File: azure/template.json (hunk 3 above)
Comment (smneal) [reply azure/template.json]: Removed serverFarmId

File: azure/template.json (hunk 4 above)
Comment (smneal) [reply azure/template.json:115]: Doing a search across the project it looks like:
- EnvironmentName is used, not Environment.
- ConfigurationStorageConnectionString is already in there
- ConfigNames is not used
- Version is not used
- APPINSIGHTS_INSTRUMENTATIONKEY is in already
- ASPNETCORE_ENVIRONMENT is in already
- LoggingRedisConnectionString is in the ConnectionStrings section as Redis

Are ConfigNames and Version something that is widely used across DAS that should be in the project?

Comment (jack-education) [reply azure/template.json:118]: All sounds good,

Having parameters for ConfigNames and Version to select the rowkey in the config table is preferable to hardcoding it like in https://github.com/SkillsFundingAgency/das-assessor-service/blob/master/src/SFA.DAS.AssessorService.Web/Startup.cs#L35-L36

Comment (smneal) [reply azure/template.json:118]: I can see that, but wouldn't it introduce the possibility of drift between what config version and name the code has set and expects v what is set during deployment?

Comment (smneal) [reply azure/template.json:126]: Those are added in.

Comment (smneal) [reply azure/template.json:126]: These have been added in.

**Pull request merged**